### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by NeverovDS**
Implement Square and Circle structures which implements
Figure interface.

### CODE REVIEW

File: figures.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+)
+
+const pi = 3.1415
```
Comment (burov) [inline figures.go]: you can use constant from ```math``` package from std

File: figures.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+type Figure interface {
+	area() float64
+	perimeter() float64
+}
+
+type Square struct {
+	a float64
```
Comment (burov) [inline figures.go]: name for this variable can be better

File: figures.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+type Circle struct {
+	r float64
+}
+
+func (b Square) perimeter() float64 {
+	b.a = 4 * b.a
```
Comment (burov) [inline figures.go]: What happens if I call this function twice ?

File: figures.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	return b.a
+}
+func (b Square) area() float64 {
+	b.a = b.a * b.a
```
Comment (burov) [inline figures.go]: What happens if I call this function twice ?

File: figures.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	return b.a
+}
+func (b Circle) area() float64 {
+	b.r = b.r * b.r * pi
```
Comment (burov) [inline figures.go]: What happens if I call this function twice ?

File: figures.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	return b.r
+}
+func (b Circle) perimeter() float64 {
+	b.r = b.r * 2 * pi
```
Comment (burov) [inline figures.go]: What happens if I call this function twice ?

**Comment by imatsiash-ib** [comment]
Please use more obvious names for the PRs.

File: figures.go (hunk 5 above)
Comment (NeverovDS) [reply figures.go]: If you call one function twice, then it will calculate the result twice

Comment (imatsiash-ib) [reply figures.go]: @NeverovDS  and with each call, the result will double, right?
This methods should not affect internal fields of structures.

File: figures.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+type Square struct {
+	side float64
+}
+type Circle struct {
+	r float64
+}
+
+func (square Square) perimeter() float64 {
+	square.side = 4 * square.side
```
Comment (imatsiash-ib) [inline figures.go]:
```suggestion
	return 4 * square.side
```

File: figures.go (hunk 5 above)
Comment (NeverovDS) [reply figures.go]: @imatsiash-ib the result dont increase in twice....

File: figures.go (hunk 7 above)
Comment (NeverovDS) [reply figures.go]: ок

File: figures.go [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+type Square struct {
+	side float64
+}
+type Circle struct {
+	r float64
+}
+
+func (square Square) perimeter() float64 {
+	perim := 4 * square.side
```
Comment (imatsiash-ib) [inline figures.go]:
```suggestion
	return 4 * square.side
```

Comment (NeverovDS) [reply figures.go]: +

**Pull request closed without merging**